docs/en_us/ingress/rewrite.md

```diff
@@ -0,0 +1,247 @@
+
+The following are allowed ways to configure the rewrite actions of url path:  
+
+* Static path
+
+* Dynamic path, including add, delete and strip path's prefix.
+
+### Static Path
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path`  to set path to specified value.
+
+For example:
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.path: '[{"params": "/index"}]'
+```
+
+Corresponding scenario:
+
+- Original url: http://host/path?query-key=value
+- Modified url: http://host/index?query-key=value
+
+### Dynamic Path
+
+Modify the prefix of url path, including add, delete and strip prefix. Be careful about setting the `order` value in the rewrite rule.
+
+#### Add Path Prefix 
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path-prefix-add` to add prefix to original path.
+
+For example：
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-add: '[{"params": "/foo/"}]'
+```
+
+Corresponding scenario:
+
+- Original url: https://host/path?query-key=value
+- Modified url:  https://host/foo/path?query-key=value
+
+#### Delete Path Prefix
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path-prefix-trim` to trim prefix from original path.
+
+For example：
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-trim: '[{"params": "/foo/"}]'
+```
+
+Corresponding scenario:
+
+* Original url: https://host/foo/path?query-key=value
+
+- Modified url: https://host/path?query-key=value
+
+#### Strip Path Prefix
+
+Use `bfe.ingress.kubernetes.io/rewrite-url.path-prefix-strip` to strip the segments of original path.
+
+For example：
+
+```yaml
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-strip: '[{"params": "1"}]'
+```
+
+Corresponding scenario:
+
+* Original url: https://host/foo/path?query-key=value
+
+- Modified url: https://host/path?query-key=value
+
+#### Order of Rewrite Rule
+
+You can define multiple annotations to modify traffic's path, but be careful about the order of rules.
+
+For example:
+
+```yaml
+# case1
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-add: '[{"params": "/index/", "order": 1}]'
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-trim: '[{"params": "/bar", "order": 2}]'
+```
+
+```yaml
+# case2
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-add: '[{"params": "/index/", "order": 2}]'
+bfe.ingress.kubernetes.io/rewrite-url.path-prefix-trim: '[{"params": "/bar", "order": 1}]'
+```
+
+The matched url is https://host/bar/other-path?query-key=value.
+
+For config of `case1` :
+
+1. Add `/index`  prefix, the url is modified to https://host/index/bar/other-path?query-key=value.
+2. Because of adding a new path prefix, the path is not matched to `/bar`  prefix, so the engine won't trim prefix. the url still is https://host/index/bar/other-path?query-key=value.
+
+ For config of `case2`：
+
+1. Because of matching to `/bar`  prefix, trim the prefix at first, then the url is modified to https://host/other-path?query-key=value.
+2. Add `/index`  prefix, the url is modified to https://host/index/other-path?query-key=value.
```

Comment:
2. Prefix `/index` will be added and the url will be rewrited to https://host/index/other-path?query-key=value.